### PR TITLE
Add workflow to generate RPM package

### PR DESCRIPTION
**Pull request by adrianfusco**
- Add **packaging** job that will build the RPM package in the workflow and publish it as an artifact
- Add the base SPEC file 

Note: keep in mind this is the base. Once other PRs are merged and we advance I'll adapt the SPEC file (e.g. add a way to generate releases using something like this https://github.com/actions/create-release )

### CODE REVIEW

**Review by cescgina — APPROVED** [review]
LGTM

File: packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+%prep
+
+%install
+echo ${GITHUB_WORKSPACE}
+
+%post
+# TODO: Uncomment and finish when this is merged: https://github.com/RedHatCRE/gz-beat-shipper/pull/2
+# systemctl enable gz-beat-shipper.service
+# systemctl start gz-beat-shipper.service
```
Comment (jsanemet) [inline packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec]: Maybe leave this up to the user. I do not know many daemons that get to start on their own.

File: packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec [hunk 2]
```diff
@@ -0,0 +1,33 @@
+%define name    gz-beat-shipper
+%define version 0.1
+%define _rpmdir %{getenv:GITHUB_WORKSPACE}/
+
+Name:           %{name}
+Version:        0.0.1
```
Comment (jsanemet) [inline packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec]: This and the one defined above do not match.

**Review by jsanemet — APPROVED** [review]
Looks good. You can take a look at my comments before merging.

File: packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec (hunk 1 above)
Comment (adrianfusco) [reply packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec]: Yes, leaving it up or not to the user are both valid. I've deleted the commented calls in the post sections.

File: packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec (hunk 2 above)
Comment (adrianfusco) [reply packing_manager_specs/RPM/rpmbuild/SPECS/gz-beat-shipper.spec]: I didn't realize, thanks. I've used the variable.

**Pull request merged**